### PR TITLE
fix: rewrite string to match package name

### LINKED ISSUE #205: `__version__` is not set in namespace packages
`ess.dream.__version__`, `ess.diffraction.__version__`, `ess.sans.__version__`, etc., are all `'0.0.0'`.

### PR DESCRIPTION
**Pull request by jokasimr**
Fixes #205

### CODE REVIEW

**Comment by jokasimr** [comment]
The issue was that '__package__` is (for example) `ess.reflectometry` but to check the version we need to pass the name of the installed package `essreflectometry` to `metadata.version`. At least that's how I understand the problem.

Does this solve the issue generally?

Tested it out in essnmx and there it resolves the issue.

File: template/src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{ projectname.removeprefix(namespace_package) }}/__init__.py.jinja [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import importlib.metadata
 
 try:
-    __version__ = importlib.metadata.version(__package__ or __name__)
+    __version__ = importlib.metadata.version((__package__ or __name__).replace("{{namespace_package}}.", "{{namespace_package}}", 1))
```
Comment (jl-wynen) [inline template/src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{ projectname.removeprefix(namespace_package) }}/__init__.py.jinja:8]: Given that this now depends on a copier variable, why not this?
```suggestion
    __version__ = importlib.metadata.version("{{projectname}}")
```

Comment (jokasimr) [reply template/src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{ projectname.removeprefix(namespace_package) }}/__init__.py.jinja:8]: Didn't know we had that variable, that's better :+1:

**Comment by jokasimr** [comment]
Do we even need to do different things in the namespace and non-namespace cases? Asking for the version of `{{projectname}}` is correct in both cases right?

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**